x/perp/keeper/grpc_query_test.go

```diff
@@ -17,6 +17,50 @@ import (
 	"github.com/NibiruChain/nibiru/x/perp/types"
 )
 
+func TestQueryAllPositions(t *testing.T) {
+	t.Log("initialize app and keeper")
+	nibiruApp, ctx := simapp.NewTestNibiruAppAndContext(true)
+	perpKeeper := &nibiruApp.PerpKeeper
+	vpoolKeeper := &nibiruApp.VpoolKeeper
+	queryServer := keeper.NewQuerier(*perpKeeper)
+
+	t.Log("initialize vpool and pair")
+	vpoolKeeper.CreatePool(
+		ctx,
+		common.Pair_BTC_NUSD,
+		/* tradeLimitRatio */ sdk.OneDec(),
+		/* quoteReserve */ sdk.MustNewDecFromStr("10000000"),
+		/* baseReserve */ sdk.MustNewDecFromStr("10000000"),
+		/* fluctuationLimitRatio */ sdk.OneDec(),
+		/* maxOracleSpreadRatio */ sdk.OneDec(),
+		/* maintenanceMarginRatio */ sdk.MustNewDecFromStr("0.0625"),
+		/* maxLeverage */ sdk.MustNewDecFromStr("15"),
+	)
+	setPairMetadata(nibiruApp.PerpKeeper, ctx, types.PairMetadata{
+		Pair: common.Pair_BTC_NUSD,
+		CumulativePremiumFractions: []sdk.Dec{
+			sdk.ZeroDec(),
+		},
+	})
+
+	for i := int64(0); i < 10; i++ {
+		setPosition(*perpKeeper, ctx, types.Position{
+			TraderAddress:                   testutil.AccAddress().String(),
+			Pair:                            common.Pair_BTC_NUSD,
+			Size_:                           sdk.MustNewDecFromStr("1000").MulInt64(i),
+			Margin:                          sdk.MustNewDecFromStr("1000").MulInt64(i),
+			OpenNotional:                    sdk.MustNewDecFromStr("1000").MulInt64(i),
+			LatestCumulativePremiumFraction: sdk.MustNewDecFromStr("1000").MulInt64(i),
+			BlockNumber:                     i,
+		})
+	}
+
+	pos, err := queryServer.QueryAllPositions(sdk.WrapSDKContext(ctx), &types.QueryAllPositionsRequest{})
```

Comment:
This test needs kind of an assertion of the positions, is not enough by just checking the error.